src/Microsoft.DotNet.Compiler.Common/AssemblyInfoFileGenerator.cs

```diff
@@ -68,9 +69,15 @@ public static string GenerateFSharp(AssemblyInfoOptions metadata)
                 [typeof(AssemblyVersionAttribute)] = EscapeCharacters(metadata.AssemblyVersion?.ToString()),
                 [typeof(AssemblyInformationalVersionAttribute)] = EscapeCharacters(metadata.InformationalVersion),
                 [typeof(AssemblyCultureAttribute)] = EscapeCharacters(metadata.Culture),
-                [typeof(NeutralResourcesLanguageAttribute)] = EscapeCharacters(metadata.NeutralLanguage),
-                [typeof(TargetFrameworkAttribute)] = EscapeCharacters(metadata.TargetFramework)
+                [typeof(NeutralResourcesLanguageAttribute)] = EscapeCharacters(metadata.NeutralLanguage)
             };
+            
+            NuGetFramework targetFramework = string.IsNullOrEmpty(metadata.TargetFramework) ? null : NuGetFramework.Parse(metadata.TargetFramework);
+            if (targetFramework != null && !(targetFramework.IsDesktop() && targetFramework.Version < new Version(4, 0)))
```

Comment:
You're right. Let me make one more suggestion (it's your call of course :))

```
        private static bool IsAllowV4Attributes(AssemblyInfoOptions metadata)
        {
            if (string.IsNullOrEmpty(metadata.TargetFramework))
            {
                // target framework is unknown. to be on the safe side, return false.
                return false;
            }

            NuGetFramework targetFramework = NuGetFramework.Parse(metadata.TargetFramework);
            if (!targetFramework.IsDesktop())
            {
                // assuming .NET Core, which should support .NET 4.0 attributes
                return true;
            }

            return targetFramework.Version >= new Version(4, 0);
        }
```
usage
```
            if (IsAllowV4Attributes(metadata))
            {
                attributes[typeof(TargetFrameworkAttribute)] = EscapeCharacters(metadata.TargetFramework); // TargetFrameworkAttribute only exists since .NET 4.0
            };
```